### DISCUSSION TITLE
Scenario-based Documentation Ideas

### DISCUSSION THREAD

**Discussion by joshooaj** [MilestoneSystemsInc/PowerShellSamples#128] [Announcements]
We get a lot of questions about how to use MilestonePSTools to complete a specific task like “How can I copy the configuration of one recording server to another, including permissions?”, or “Can I change all cameras using driver X over to new driver Y, enable the new hardware analytic event and create a rule to record on that event?”

We have a fair few examples we can dig up from past emails but I would also like to hear from you about what scenarios you think should be documented at [www.milestonepstools.com](www.milestonepstools.com).

The kind of scenarios I have in mind are things that could be done in up to ~20-30 lines of code - enough to demonstrate how you might use a several commands together, but not so much that the example itself becomes a tool to be maintained in its own right. That said, I’d appreciate any and all ideas you have - what would you like to see in the docs?

**Comment by joshooaj** [comment]
Here are a few ideas that have come direct to me - sharing here so I have one place to reference later:

- What about Export-VmsHardware and Import-VmsHardware in an example for how to migrate hardware from one Management Server to another - like when you need to merge many Management Servers into one for example.
- Exporting and importing roles and users for the same purpose as above.
- Email a daily totals report for cameras with people/car/whatever counting analytics.  
  ![image](https://github.com/MilestoneSystemsInc/PowerShellSamples/assets/7669972/cbd23a98-3ef0-4019-b6c5-890fe4ed6221)
- How to publish custom VMS metrics in [OpenMetrics](https://github.com/OpenObservability/OpenMetrics) format compatible with monitoring solutions like [Prometheus](https://prometheus.io/).
- How to create generic events in bulk and corresponding rules or alarms.

**Comment by Tarterman** [comment]
Enabling all metadata channels (or maybe all metadata channels for cameras in a given device group) and add those metadata channels to a device group.
